cli/main.go

```diff
@@ -28,6 +37,99 @@ func main() {
+			{
+				Name:  "status",
+				Usage: "Checks the status of your eigenpod.",
+				Flags: []cli.Flag{
+					&cli.BoolFlag{
+						Name:        "json",
+						Value:       false,
+						Usage:       "print only plain JSON",
+						Required:    false,
+						Destination: &useJson,
+					},
+				},
+				Action: func(cctx *cli.Context) error {
+					if disableColor {
+						color.NoColor = true
+					}
+
+					eth, err := ethclient.Dial(node)
+					PanicOnError("failed to reach eth --node.", err)
+
+					beaconClient, err := getBeaconClient(beacon)
+					PanicOnError("failed to reach beacon chain.", err)
+
+					status := getStatus(ctx, eigenpodAddress, eth, beaconClient)
+
+					if useJson {
+						bytes, err := json.MarshalIndent(status, "", "      ")
+						PanicOnError("failed to get status", err)
+						statusStr := string(bytes)
+						fmt.Println(statusStr)
+					} else {
+						// pretty print everything
+						color.New(color.Bold, color.FgBlue).Printf("Eigenpod validators\n")
+						for index, validator := range status.Validators {
+
+							var targetColor color.Attribute
+							var description string
+
+							if validator.Status == ValidatorStatusActive {
+								targetColor = color.FgGreen
+								description = "active"
+							} else if validator.Status == ValidatorStatusInactive {
+								targetColor = color.FgHiYellow
+								description = "inactive"
+							} else if validator.Status == ValidatorStatusWithdrawn {
+								targetColor = color.FgHiRed
+								description = "withdrawn"
+							}
+
+							if validator.Slashed {
+								description = description + " (slashed)"
+							}
+
+							publicKey := validator.PublicKey
+							if !verbose {
+								publicKey = shortenHex(publicKey)
+							}
+
+							color.New(targetColor).Printf("\t- #%s (%s) [%s]\n", index, publicKey, description)
+						}
+
+						bold := color.New(color.Bold, color.FgBlue)
+						ital := color.New(color.Italic, color.FgBlue)
+						fmt.Println()
+
+						if status.ActiveCheckpoint != nil {
+							startTime := time.Unix(int64(status.ActiveCheckpoint.StartedAt), 0)
+
+							bold.Printf("!NOTE: There is a checkpoint active! (started at: %s)\n", startTime.String())
+
+							endSharesETH := gweiToEther(status.ActiveCheckpoint.PendingSharesGwei)
+							deltaETH := new(big.Float).Sub(
+								endSharesETH,
+								status.CurrentTotalSharesETH,
+							) // delta = endShares - currentOwnerSharesETH
+
+							ital.Printf("\t- If you finish it, you may receive up to %s shares. (%s -> %s)\n", deltaETH.String(), status.CurrentTotalSharesETH.String(), endSharesETH.String())
+
+							ital.Printf("\t- %d proof(s) remaining until completion.\n", status.ActiveCheckpoint.ProofsRemaining)
+						} else {
+							bold.Printf("Runing a `checkpoint` right now will result in: \n")
+
+							startEther := status.CurrentTotalSharesETH
+							endEther := status.TotalSharesAfterCheckpointETH
+							delta := new(big.Float).Sub(endEther, startEther)
```

Comment:
Might be worth refactoring so that `TotalSharesAfterCheckpointETH` is used whether you have a checkpoint or not -- since right now you're reading from either this or `PendingSharesGwei` and so the logic is a little bit distributed between `status.go` and `main.go`.